lib/core/test/unit/Cardano/Pool/MetricsSpec.hs

```diff
@@ -62,6 +69,28 @@ spec = do
     describe "calculatePerformances" $ do
         it "performances are always between 0 and 1"
             $ property prop_performancesBounded01
+        it "50% stake, producing 4/8 blocks => performance=1" $ do
+           let p = calculatePerformance 8 stake (productions 4)
+           Map.lookup pool p `shouldBe` (Just 1)
+        it "50% stake, producing 2/8 blocks => performance=0.5" $ do
+           let p = calculatePerformance 8 stake (productions 2)
+           Map.lookup pool p `shouldBe` (Just 0.5)
+        it "50% stake, producing 0/8 blocks => performance=0" $ do
+           let p = calculatePerformance 8 stake (productions 0)
+           Map.lookup pool p `shouldBe` (Just 0)
```

Comment:
It'd be clearer from a reader perspective for `stake` and `productions` to actually take arguments and be defined in the test scenarios themselves. For example:

```hs
       it "50% stake, producing 4/8 blocks => performance=1" $ do
           let stake = mkStake [(poolA, 1), (poolB, 1)]
           let production  = mkProduction [(poolA, 4), (poolB, 0)]
           let performances = calculatePerformance 8 stake production
           Map.lookup poolA performances `shouldBe` (Just 1)
```

doesn't need any surrounding context to be read and understood. Plus, it prevents a change in `stake` or `productions` for a given test to impact other tests :+1: